lib/systems/architectures.nix

```diff
@@ -37,17 +42,34 @@ rec {
     loongson2f     = [ ];
   };
 
+  # map gcc.arch to nix' microarchitecture level system types
+  map-microarchitecture-levels = {
+    westmere       = "x86_64-v2";
+    sandybridge    = "x86_64-v2";
+    ivybridge      = "x86_64-v2";
+    haswell        = "x86_64-v3";
+    broadwell      = "x86_64-v3";
+    skylake        = "x86_64-v3";
+    skylake-avx512 = "x86_64-v4";
+    cannonlake     = "x86_64-v4";
+    icelake-client = "x86_64-v4";
+    icelake-server = "x86_64-v4";
+    cascadelake    = "x86_64-v4";
+    cooperlake     = "x86_64-v4";
+    tigerlake      = "x86_64-v4";
+  };
```

Comment:
I am not sure if we need those now at all after I added the microarchitecture levels to inferiors

Reply:
After realizing that we need those in inferiors there is no longer a usecase for this.